samcli/local/docker/container.py

```diff
@@ -402,7 +402,7 @@ def wait_for_http_response(self, name, event, stdout) -> Tuple[Union[str, bytes]
 
         try:
             # if response is an image then json.loads/dumps will throw a UnicodeDecodeError so return raw content
-            if "image" in resp.headers["Content-Type"]:
+            if resp.headers.get("Content-Type") and "image" in resp.headers["Content-Type"]:
```

Comment:
```suggestion
            if "image" in resp.headers.get("Content-Type"):
```
This can be done in one check as .get() returns None if not present. This means the first check is redundant.

Reply:
I think we still need to check for the existence of the `Content-Type` field since otherwise the if statement boils down to `if "image" in None` which throws an error

Reply:
Apologies your correct, the fix I added locally for this defaulted it